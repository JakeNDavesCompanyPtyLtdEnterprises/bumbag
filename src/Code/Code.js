// @flow
import React, { type Node } from 'react';

import _Code from './styled';

type Props = {
  children: Node,
  className?: string,
  codeClassName?: string,
  isBlock?: boolean
};

const Code = ({ children, className, codeClassName, isBlock, ...props }: Props) => (
  <_Code className={className} codeClassName={codeClassName} block={isBlock} {...props}>
    {children}
  </_Code>
);

Code.defaultProps = {
  className: undefined,
  codeClassName: undefined,
  isBlock: false
};

export default Code;
