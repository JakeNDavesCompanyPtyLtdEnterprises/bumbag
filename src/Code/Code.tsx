import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import { CodeProps as ReakitCodeProps } from 'reakit/ts';

import _Code from './styled';

export interface LocalCodeProps {
  children: React.ReactNode;
  className?: string;
  codeClassName?: string;
  isBlock?: boolean;
}
export type CodeProps = LocalCodeProps & ReakitCodeProps;

export const Code: React.FunctionComponent<LocalCodeProps> = ({
  children,
  className,
  codeClassName,
  isBlock,
  ...props
}) => (
  <_Code block={isBlock} {...props}>
    {children}
  </_Code>
);

Code.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  codeClassName: PropTypes.string,
  isBlock: PropTypes.bool
};
Code.defaultProps = {
  className: undefined,
  codeClassName: undefined,
  isBlock: false
};

const C: React.FunctionComponent<CodeProps> = Code;
export default C;
