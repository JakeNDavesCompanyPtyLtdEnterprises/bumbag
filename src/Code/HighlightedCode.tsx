import * as React from 'react';
// @ts-ignore
import SyntaxHighlighter, { registerLanguage } from 'react-syntax-highlighter/prism-light';
// @ts-ignore
import defaultTheme from 'react-syntax-highlighter/styles/prism/atom-dark';
// @ts-ignore
import javascript from 'react-syntax-highlighter/languages/prism/javascript';
// @ts-ignore
import jsx from 'react-syntax-highlighter/languages/prism/jsx';
import { Code as _Code } from 'reakit';
import { BoxProps as ReakitBoxProps } from 'reakit/ts';

import { Box } from '../primitives';
import { styled, css } from '../styled';

const Wrapper = styled(Box)<HighlightedCodeProps>`
  ${props =>
    props.showLabel &&
    css`
  position: relative;

  &:before {
    color: #e4e4e4;
    content: '${props => props.lang}';
    font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
    font-size: 0.8rem;
    font-weight: bold;
    position: absolute;
    right: 0.6rem;
    top: 0.2rem;
    text-transform: uppercase;
  }
  `};
`;

export interface LocalHighlightedCodeProps {
  children: React.ReactNode;
  className?: string;
  codeClassName?: string;
  isBlock?: boolean;
  lang: string;
  showLabel?: boolean;
  showLineNumbers?: boolean;
}
export type HighlightedCodeProps = LocalHighlightedCodeProps & ReakitBoxProps;

export class HighlightedCode extends React.PureComponent<LocalHighlightedCodeProps> {
  static defaultProps = {
    className: undefined,
    codeClassName: undefined,
    isBlock: false,
    showLabel: false,
    showLineNumbers: false
  };

  componentDidMount = () => {
    registerLanguage('javascript', javascript);
    registerLanguage('jsx', jsx);
  };

  render = () => {
    const { children, className, codeClassName, isBlock, lang, showLineNumbers, ...props } = this.props;
    return (
      <Wrapper lang={lang} {...props}>
        <SyntaxHighlighter
          className={className}
          codeClassName={codeClassName}
          block={isBlock}
          language={lang}
          showLineNumbers={showLineNumbers}
          style={defaultTheme}
          PreTag={_Code}
        >
          {children}
        </SyntaxHighlighter>
      </Wrapper>
    );
  };
}

export default HighlightedCode as React.ComponentClass<HighlightedCodeProps>;
