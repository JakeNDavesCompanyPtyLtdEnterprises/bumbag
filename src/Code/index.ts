export { default as Code } from './Code';
export { default as HighlightedCode } from './HighlightedCode';
export { default } from './Code';
