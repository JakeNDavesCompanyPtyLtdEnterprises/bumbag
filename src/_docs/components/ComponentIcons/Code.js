import React from 'react';

const SvgComponent = props => (
  <svg width={54} height={21} fill="none" {...props}>
    <path
      d="M.8 14.52V9.48L15.2 3v4.5L4.52 12l10.68 4.5V21L.8 14.52zM30.864.36h5.82L28.704 21h-5.82L30.864.36zm7.934 16.14l10.68-4.5-10.68-4.5V3l14.4 6.48v5.04L38.798 21v-4.5z"
      fill="#444bc9"
    />
  </svg>
);

export default SvgComponent;
