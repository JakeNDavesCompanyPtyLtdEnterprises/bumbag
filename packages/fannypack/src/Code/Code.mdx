---
name: Code
route: /typography/code
menu: Typography
---

import { Playground } from 'docz';
import PropsTable from '../_docs/components/PropsTable';
import { Box } from '../primitives/index';
import { Code } from './Code';
import { HighlightedCode } from './HighlightedCode';

# Code

## Import

`import { Code, HighlightedCode } from 'fannypack'`

## Basic Usage

`<Code>` renders a `<code>` element by default with reset styles and theme styles.

<Playground>
  <Code>Hello world!</Code>
</Playground>

## Code blocks

A code block can be rendered by setting the `isBlock` prop. This will wrap the code in a `<pre>`.

<Playground>
  <Code isBlock>
  {`function log(message) {
    console.log(message);
  }`}
  </Code>
</Playground>

## Highlighted code

The `<HighlighedCode>` component uses [Prism](https://prismjs.com/) under the hood. To highlight a language, specify the `lang` prop.

<Playground>
  <HighlightedCode lang="javascript">
  {`console.log('Hello world')`}
  </HighlightedCode>
  <HighlightedCode isBlock lang="html" marginTop="major-3">
  {`<body>
    <div class="content">Hello world</div>
  </body>`}
  </HighlightedCode>
</Playground>

You can also pass a `showLabel` prop to show a label of the language in the top right corner:

<Playground>
  <HighlightedCode isBlock lang="javascript" showLabel>
  {`function log(message) {
    console.log(message);
  }`}
  </HighlightedCode>
</Playground>

and show line numbers with `showLineNumbers`:

<Playground>
  <HighlightedCode isBlock lang="javascript" showLineNumbers>
  {`function log(message) {
    console.log(message);
  }`}
  </HighlightedCode>
</Playground>



## `<Code>` Props

<PropsTable component={Code} />

## `<HighlightedCode>` Props

<PropsTable component={HighlightedCode} />

## Theming

### Schema

```jsx
{
  Code: {
    base: css | Object,

    // Styles to apply for blocked code
    block: css | Object,

    // Styles to apply for inline code
    inline: css | Object
  }
}
```

[Learn more about theming](/styling#theming)
