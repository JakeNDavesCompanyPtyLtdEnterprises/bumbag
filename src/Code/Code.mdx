---
name: Code
route: /typography/code
menu: Typography
---

import { Playground, PropsTable } from 'docz';
import { Box } from '../primitives/index';
import { Code } from './index';
import { HighlightedCode } from './HighlightedCode';

# Code

## Import

`import { Code, HighlightedCode } from 'fannypack'`

## Basic Usage

`<Code>` renders a `<code>` element by default with reset styles and theme styles.

<Playground>
  <Code>Hello world!</Code>
</Playground>

## Code blocks

A code block can be rendered by setting the `isBlock` prop. This will wrap the code in a `<pre>`.

<Playground>
  <Code isBlock>
  {`function log(message) {
    console.log(message);
  }`}
  </Code>
</Playground>

## Highlighted code

The `<HighlighedCode>` component uses [Prism](https://prismjs.com/) under the hood. To highlight a language, specify the `lang` prop.

<Playground>
  <HighlightedCode lang="javascript">
  {`console.log('Hello world')`}
  </HighlightedCode>
  <HighlightedCode isBlock lang="html" marginTop="medium">
  {`<body>
    <div class="content">Hello world</div>
  </body>`}
  </HighlightedCode>
</Playground>

You can also pass a `showLabel` prop to show a label of the language in the top right corner:

<Playground>
  <HighlightedCode isBlock lang="javascript" showLabel>
  {`function log(message) {
    console.log(message);
  }`}
  </HighlightedCode>
</Playground>

and show line numbers with `showLineNumbers`:

<Playground>
  <HighlightedCode isBlock lang="javascript" showLineNumbers>
  {`function log(message) {
    console.log(message);
  }`}
  </HighlightedCode>
</Playground>



## `<Code>` Props

<PropsTable of={Code} />

## `<HighlightedCode>` Props

<PropsTable of={HighlightedCode} />

## Theming

### Schema

```jsx
{
  base: string | Object,
  block: string | Object,
  inline: string | Object
}
```
