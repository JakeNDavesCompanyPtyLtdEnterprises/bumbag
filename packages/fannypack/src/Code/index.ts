import Code from './Code';
import HighlightedCode from './HighlightedCode';

export { Code, HighlightedCode };
export default Code;
