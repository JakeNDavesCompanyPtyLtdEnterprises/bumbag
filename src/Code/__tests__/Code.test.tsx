import * as React from 'react';
import render from '../../_utils/tests/render';
import Code from '../Code';

it('renders correctly for code (inline)', () => {
  const { container } = render(<Code>test</Code>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for code (block)', () => {
  const { container } = render(<Code isBlock>test</Code>);
  expect(container.firstChild).toMatchSnapshot();
});
