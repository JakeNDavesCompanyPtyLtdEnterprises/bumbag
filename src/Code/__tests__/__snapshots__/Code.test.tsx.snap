// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for code (block) 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  background-color: #f2f2f2;
  border-radius: 0.1rem;
  color: #435a6f;
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

code.c0 {
  padding: 0.1rem 0.2rem;
}

pre.c0 {
  padding: 1rem;
}

<pre
  class="c0 c1"
>
  <code>
    test
  </code>
</pre>
`;

exports[`renders correctly for code (inline) 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  background-color: #f2f2f2;
  border-radius: 0.1rem;
  color: #435a6f;
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

code.c0 {
  padding: 0.1rem 0.2rem;
}

pre.c0 {
  padding: 1rem;
}

<code
  class="c0 c1"
>
  test
</code>
`;
